src/platform/libretro/libretro.c

```diff
@@ -450,64 +549,97 @@ bool retro_load_game(const struct retro_game_info* game)
+	core1->setAVStream(core1, &stream1);
+	mCoreInitConfig(core2, NULL);
+	core2->init(core2);
+	core2->setAVStream(core2, &stream2);
+	mCoreInitConfig(core3, NULL);
+	core3->init(core3);
+	core3->setAVStream(core3, &stream3);
+	mCoreInitConfig(core4, NULL);
+	core4->init(core4);
+	core4->setAVStream(core4, &stream4);
 #ifdef _3DS
 	outputBuffer = linearMemAlign(256 * VIDEO_VERTICAL_PIXELS * BYTES_PER_PIXEL, 0x80);
 #else
-	outputBuffer = malloc(256 * VIDEO_VERTICAL_PIXELS * BYTES_PER_PIXEL);
+	outputBuffer  = (uint8_t*)malloc(1024 * VIDEO_VERTICAL_PIXELS * BYTES_PER_PIXEL);
 #endif
-	core->setVideoBuffer(core, outputBuffer, 256);
-
-	core->setAudioBufferSize(core, SAMPLES);
+	core1->setVideoBuffer(core1, outputBuffer, 512);
+	core2->setVideoBuffer(core2, (uint8_t*)outputBuffer + BYTES_PER_PIXEL * VIDEO_HORIZONTAL_PIXELS, 512);
+	core3->setVideoBuffer(core3, (uint8_t*)outputBuffer + BYTES_PER_PIXEL * (512 * VIDEO_VERTICAL_PIXELS), 512);
+	core4->setVideoBuffer(core4, (uint8_t*)outputBuffer + BYTES_PER_PIXEL * (512 * VIDEO_VERTICAL_PIXELS + VIDEO_HORIZONTAL_PIXELS), 512);
 
-	blip_set_rates(core->getAudioChannel(core, 0), core->frequency(core), 32768);
-	blip_set_rates(core->getAudioChannel(core, 1), core->frequency(core), 32768);
+	blip_set_rates(core1->getAudioChannel(core1, 0), core1->frequency(core1), 32768);
+	blip_set_rates(core1->getAudioChannel(core1, 1), core1->frequency(core1), 32768);
 
-	core->setPeripheral(core, mPERIPH_RUMBLE, &rumble);
+   core1->setPeripheral(core1, mPERIPH_RUMBLE, &rumble);
 
 	savedata = anonymousMemoryMap(SIZE_CART_FLASH1M);
 	struct VFile* save = VFileFromMemory(savedata, SIZE_CART_FLASH1M);
 
 	_reloadSettings();
-	core->loadROM(core, rom);
-	core->loadSave(core, save);
+	core1->loadROM(core1, rom);
+	core1->loadSave(core1, save);
+	core2->loadROM(core2, rom);
+	core2->loadSave(core2, save);
+	core3->loadROM(core3, rom);
+	core3->loadSave(core3, save);
+	core4->loadROM(core4, rom);
+	core4->loadSave(core4, save);
+	gba1 = core1->board;
+	GBASIOInit(&gba1->sio);
 
 #ifdef M_CORE_GBA
-	if (core->platform(core) == PLATFORM_GBA) {
-		core->setPeripheral(core, mPERIPH_GBA_LUMINANCE, &lux);
+	if (core1->platform(core1) == PLATFORM_GBA) {
+		struct GBA* gba = core1->board;
+		gba->luminanceSource = &lux;
 
 		const char* sysDir = 0;
-		if (core->opts.useBios && environCallback(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &sysDir)) {
+		if (core1->opts.useBios && environCallback(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &sysDir)) {
 			char biosPath[PATH_MAX];
 			snprintf(biosPath, sizeof(biosPath), "%s%s%s", sysDir, PATH_SEP, "gba_bios.bin");
 			struct VFile* bios = VFileOpen(biosPath, O_RDONLY);
 			if (bios) {
-				core->loadBIOS(core, bios, 0);
+				core1->loadBIOS(core1, bios, 0);
+				core2->loadBIOS(core2, bios, 0);
+				core3->loadBIOS(core3, bios, 0);
+				core4->loadBIOS(core4, bios, 0);
 			}
 		}
 	}
 #endif
 
-	core->reset(core);
-	_setupMaps(core);
-
+	core1->reset(core1);
+	_setupMaps(core1);
```

Comment:
wait, why is this called from two different places, is that really correct

Reply:
isn't this retro_load_game? I'm just following the original core here